qrexec/tests/socket/daemon.py

```diff
@@ -268,6 +287,58 @@ def recv_refused(agent):
             )
             recv_refused(agent)
 
+    def test_new_style_request(self):
+        """
+        Test that qrexec-daemon accepts request.
+        """
+        agent = self.start_daemon_with_agent()
+        agent.handshake()
+
+        target_domain_name = "target_domain"
+        ident = "ab"
+
+        # check policy program output
+        policy_program_path = os.path.join(self.tempdir, "qrexec-policy-exec")
+
+        # set deny
+        self.set_policy_params(1, 1)
+
+        result = subprocess.run(
+            [policy_program_path, "--", "somedomain", "anotherdomain", "someservice"],
+            capture_output=True,
+            text=True
+        )
+        assert result.stdout == "result=deny\n"
+
+        # set allow
+        self.set_policy_params(1, 0)
+
+        result = subprocess.run(
+            [policy_program_path, "--", "somedomain", "anotherdomain", "someservice"],
+            capture_output=True,
+            text=True
+        )
+        assert result.stdout == """result=allow
+autostart=True
+user=toto
+target=somedomain
+requested_target=anotherdomain
+"""
+
+        # check allowed request
+        agent.send_message(
+            qrexec.MSG_TRIGGER_SERVICE3,
+            struct.pack("<64s32s", target_domain_name.encode(), ident.encode())
+            + b"a\0",
+        )
+        message_type, data = agent.recv_message()
```

Comment:
I guess this only works because your policy redirected the request back to the source domain (see earlier comment). For this to really work, you should have `target_domain_name = self.domain_name` (real policy will refuse loopback connections, but fortunately you don't have real policy here :) ).